src/Unosquare.PassCore.Web/Controllers/PasswordController.cs

```diff
@@ -43,6 +52,42 @@ public class PasswordController : Controller
         [HttpGet]
         public IActionResult Get() => Json(_options);
 
+        /// <summary>
+        /// Returns generated password as a JSON string.
+        /// </summary>
+        /// <returns>A Json with a password property which contains a random generated password.</returns>
+        [HttpGet]
+        [Route("generated")]
+        public IActionResult GetGeeneratedPassword()
+        {
+            var score = 0;
+            var password = string.Empty;
+            var maxNumber = _words.Length;
+
+            do
+            {
+                var firstIndex = _random.Next(0, maxNumber);
+                var secondIndex = _random.Next(0, maxNumber);
+
+                // if both indexes have the same value, repeat the random
+                if (secondIndex == firstIndex) continue;
+
+                var firstWord = _words[firstIndex];
+                if (int.Parse(firstIndex.ToString().Substring(0, 1)) < 4)
+                    firstWord = char.ToUpper(firstWord[0], CultureInfo.InvariantCulture) + firstWord.Substring(1);
+
+                var secondWord = _words[secondIndex];
+                if (int.Parse(secondIndex.ToString().Substring(0, 1)) < 4)
+                    secondWord = char.ToUpper(secondWord[0], CultureInfo.InvariantCulture) + secondWord.Substring(1);
+
+                password = $"{firstWord}{secondIndex}_{secondWord}{firstIndex}";
+                score = Zxcvbn.MatchPassword(password).Score;
+            }
+            while (score < 4);

```

Comment:
Add a new setting with the minimum score required.

Reply:
You shouldn't check the score. Instead encode a specified number of bits

Reply:
@graingert can you explain to us more details about the encoding?